cmd/helm-migrate-values/root.go

```diff
@@ -124,26 +133,38 @@ func newRunner(actionConfig *action.Configuration, flags *pflag.FlagSet, setting
 			relMajorVer, _ := strconv.Atoi(matches[1])
 			migrationsPath := filepath.Join(*chartDir, release.Chart.Name(), "value-migrations")
 
+			// vTo is always nil, because it really only makes sense to migrate to the current chart version.
+			// The migrations library does support migrating to a specific version though.
 			migratedConfig, err := pkg.MigrateFromPath(release.Config, relMajorVer, nil, migrationsPath, log)
 			if err != nil {
 				return err
 			}
 
-			migratedValues, err := yaml.Marshal(migratedConfig)
-			if err != nil {
-				return fmt.Errorf("migrated values are in an invalid format: %w", err)
-			}
+			if len(migratedConfig) > 0 {
 
-			if *outputFile == "" {
-				message := fmt.Sprintf("Migrated values for release %s:\n%s", name, string(migratedValues))
-				if _, err = fmt.Fprint(out, message); err != nil {
-					return fmt.Errorf("error writing migrated values to standard output: %w", err)
+				migratedValues, err := yaml.Marshal(migratedConfig)
+				if err != nil {
+					return fmt.Errorf("migrated values are in an invalid format: %w", err)
 				}
-			} else {
-				if err = writeOutputValues(err, *outputFile, migratedValues); err != nil {
-					return err
+
+				if *outputFile == "" {
+					message := fmt.Sprintf("Migrated user-supplied values for release %s:\n%s", name, string(migratedValues))
+					if _, err = fmt.Fprint(out, message); err != nil {
+						return fmt.Errorf("error writing migrated values to standard output: %w", err)
+					}
+				} else {
+					if err = writeOutputValues(err, *outputFile, migratedValues); err != nil {
+						return err
+					}
 				}
+			} else {
+				migrationRequired = false
 			}
+
+		}
+
+		if !migrationRequired {
+			log.Warning("No migration required for release %s", name)
```

Comment:
If you want this message to go to stdout to be more obvious, then you can add a new method to the Logger in `logging.go` called something like `Info`. This will be similar to warning except it would format with `"INFO: %s\n" and write to stdout instead of stderr.

I think this is preferred over free-form printf because at least it makes it obvious which messages are logs and which aren't.

Reply:
I might just add that for no migration needed, but leave it as a warning for no migrations found.